### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by hafiz1379**
### Summary
This PR introduces three classes: `Classroom`, `Book`, and `Rental`, along with the implementation of their respective relationships. Additionally, it establishes a many-to-many (has-many-through) relationship between `Person` and `Book` using the `Rental` intermediate class. The project fulfills all the specified requirements and enhances the structure of managing classrooms, books, and rentals.

### Changes Made

**Classroom Class**
- Created a `Classroom` class with an `@label` instance variable, initialized in the constructor.
- Implemented setters and getters for `@label` using `attr_accessor`.
- Established the has-many relationship between `Classroom` and `Student`.
- Implemented methods to add a student to a classroom and set the classroom for a student.

**Book Class**
- Created a `Book` class with `@title` and `@author` instance variables, initialized in the constructor.
- Implemented setters and getters for `@title` and `@author` using `attr_accessor`.

**Rental Class**
- Created a `Rental` class with an `@date` instance variable, initialized in the constructor.
- Implemented setters and getters for `@date` using `attr_accessor`.
- Established the many-to-many (has-many-through) relationship between `Person` and `Book` using `Rental`.
- Implemented the has-many side for `Book` and `Rental` and the belongs-to side for `Rental` and `Book`.
- Created the has-many side for `Person` and `Rental` and the belongs-to side for `Rental` and `Person`.
- Modified the constructor of `Rental` to set the `Book` and `Person` instances.

### Reviewer's Notes
I welcome the code reviewer to examine the implementation, test cases, and relationships between classes. Please provide feedback on code quality, design, and correctness. Your insights are highly appreciated. Thank you for taking the time to review this PR.

### CODE REVIEW

File: Lib/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classroom = self
+  end
+end
```
Comment (danmainah) [inline Lib/classroom.rb:13]: - [ ] @hafiz1379 Great work adding the classroom class and adding the 'has-many' side of the relationship between class and students, however, you forgot to add the 'belongs-to' side of the relationship in your student class, kindly consider adding it.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @hafiz1379 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Created book and rental classes and ensured the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental is implemented.
✔️  Created classroom class and ensured the has-many side (a classroom has many students) relationship has been implemented.


# Required Changes ♻️

- [ ] @hafiz1379 It would have been nice if you completed the commands for setting up and  running your project in the getting started section, currently your readme looks incomplete due to the incomplete statements as shown below:
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: Lib/student.rb [hunk 2]
```diff
@@ -6,8 +6,16 @@ class Student < Person
...
+    classroom&.add_student(self)
   end
 
+  # rubocop:disable Naming/AccessorMethodName
+  # Add a method to get the classroom to which the student belongs.
+  def get_classroom
+    @classroom
+  end
+  # rubocop:enable Naming/AccessorMethodName
+
```
Comment (Strangeal) [inline Lib/student.rb]: - [ ] Great job on your progress in addressing the feedback provided by the previous reviewer! 👏 However, I'd like to highlight some adjustments that are necessary for your code to align with the project requirements.

   - As per the project requirements, you should establish a bidirectional association between the `student` and `classroom` classes. This means that when you set the `classroom` for a `student`, it should also automatically add the student to the classroom's list of students. You can refer to this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) to gain a deeper understanding of this concept.

To achieve this association, ensure that your method for setting the classroom for a student looks like this:

```ruby
  def assign_classroom(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self) 
  end
```

By implementing this bidirectional association, your code will meet the project requirements and maintain a more intuitive relationship between students and classrooms.

Happy coding! 😊🚀🍀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @hafiz1379, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text

 ## Highlights
- Added `Getting started` info in `README`

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Lib/classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (DeliceLydia) [inline Lib/classroom.rb:2]: ## Optional
- [ ] I suggest that you make `students` an `attr_reader` only

**Review by DeliceLydia — APPROVED** [review]
## Hi @hafiz1379, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

## Highlights
- Addressed what previous reviewer suggested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**